query-node/mappings/src/workingGroups.ts

```diff
@@ -335,27 +335,40 @@ async function handleWorkingGroupMetadataAction(
 
 async function handleTerminatedWorker({ store, event }: EventContext & StoreContext): Promise<void> {
   const [workerId, optPenalty, optRationale] = new WorkingGroups.TerminatedWorkerEvent(event).params
-  const group = await getWorkingGroup(store, event)
-  const worker = await getWorker(store, group.name as WorkingGroupModuleName, workerId, ['application'])
+  const group = await getWorkingGroup(store, event, ['leader'])
+  const worker = await getWorker(store, group.name as WorkingGroupModuleName, workerId, [
+    'application',
+    'workinggroupleader',
+  ])
 
-  const EventConstructor = worker.isLead ? TerminatedLeaderEvent : TerminatedWorkerEvent
+  if (worker.isLead) {
+    const terminatedLeaderEvent = new TerminatedLeaderEvent({
+      ...genericEventFields(event),
+      group,
+      worker,
+      penalty: optPenalty.unwrapOr(undefined),
+      rationale: optRationale.isSome ? bytesToString(optRationale.unwrap()) : undefined,
+    })
+    await store.save<TerminatedLeaderEvent>(terminatedLeaderEvent)
+  }
 
-  const terminatedEvent = new EventConstructor({
+  const terminatedWorkerEvent = new TerminatedWorkerEvent({
     ...genericEventFields(event),
     group,
     worker,
     penalty: optPenalty.unwrapOr(undefined),
     rationale: optRationale.isSome ? bytesToString(optRationale.unwrap()) : undefined,
   })
 
-  await store.save(terminatedEvent)
+  await store.save<TerminatedWorkerEvent>(terminatedWorkerEvent)
 
   const status = new WorkerStatusTerminated()
-  status.terminatedWorkerEventId = terminatedEvent.id
```

Comment:
@zeeshanakram3 about [the comment on `TerminatedLeaderEvent` & `TerminatedWorkerEvent` entites](https://github.com/Joystream/joystream/pull/4469/files#r1040627135):

Here either `TerminatedLeaderEvent` or `TerminatedWorkerEvent` are passed to the `WorkerStatusTerminated` variant even though it expects `TerminatedWorkerEvent` exclusively:
```gql
type WorkerStatusTerminated @variant {
  "Related event emitted on worker termination"
  terminatedWorkerEvent: TerminatedWorkerEvent!
}
```
(plus isn't the ids of these 2 events non mutually unique ?)
I'm confuse about this but if this is an error I see is 3 solutions:
1. add a variant into the `WorkerStatus` union but this breaks the code which check for `WorkerStatusTerminated` for terminated leads (also this would be the only lead specific status).
2. make `WorkerStatusTerminated.terminatedWorkerEvent` into a union of variant this one is better IMO but it's  also a breaking change (although I'm not sure how much change is needed maybe it's not too much).
3. emitting `TerminatedLeaderEvent` along with `TerminatedWorkerEvent` but as you mentioned it differs from the runtime.

I had not realized that `3` was inconsistent with the runtime working, however I'm not sure if it's bad enough to go for `1` or `2` or maybe I'm missing something.

WDYT @zeeshanakram3 ?